overlord/state/change.go

```diff
@@ -282,32 +291,119 @@ func init() {
 	}
 }
 
+func (c *Change) isTaskWaiting(visited map[string]taskWaitComputeStatus, t *Task, deps []*Task) bool {
+	taskID := t.ID()
+	computeStatus := visited[taskID]
+	switch computeStatus {
+	case taskWaitStatusComputing:
+		// Cyclic dependency detected, return false to short-circuit.
+		return false
+	case taskWaitStatusWaiting, taskWaitStatusNotWaiting:
+		return computeStatus == taskWaitStatusWaiting
+	}
+	visited[taskID] = taskWaitStatusComputing
+
+	var isWaiting bool
+depscheck:
```

Comment:
This label isn't needed, we could just have a `break`s with no labels

Reply:
It is needed, the `break` is inside a switch statement. The break will only break out of the switch, not the for loop.

Reply:
My bad, I saw wrong